src/System.Private.CoreLib/src/System/Text/StringBuilder.CoreCLR.cs

```diff
@@ -29,7 +29,10 @@ internal unsafe void ReplaceBufferInternal(char* newBuffer, int newLength)
 
             if (newLength > m_ChunkChars.Length)
             {
-                m_ChunkChars = new char[GetReplaceBufferCapacity(newLength)];
+                // m_ChunkChars = new char[GetReplaceBufferCapacity(newLength)];
+
+                // DO NOT MERGE: for testing only 
+                m_ChunkChars = System.GC.AllocateUninitializedArray<char>(GetReplaceBufferCapacity(newLength));
```

Comment:
Is the idea that we'd replace pretty much every internal array allocation in corelib with this call?

Reply:
You would need to measure before making such changes for real. I would assume that some types will benefit some will not.

The prerequisite is to not rely on initial state of the buffer be zeroed. `List<T>` and `StringBuilder` fit such pattern, so they _can_ use this API. If they can`t then we have a bug and I want to know about it.

Whether the change is actually beneficial - that is a different story.